wls-ergebnismeldung-service/src/main/java/de/muenchen/oss/wahllokalsystem/ergebnismeldungservice/client/basisdaten/BasisdatenClient.java

```diff
@@ -0,0 +1,78 @@
+package de.muenchen.oss.wahllokalsystem.ergebnismeldungservice.client.basisdaten;
+
+import de.muenchen.oss.wahllokalsystem.ergebnismeldungservice.configuration.Profiles;
+import de.muenchen.oss.wahllokalsystem.ergebnismeldungservice.eai.basisdaten.client.WahlenControllerApi;
+import de.muenchen.oss.wahllokalsystem.ergebnismeldungservice.eai.basisdaten.model.WahlDTO;
+import de.muenchen.oss.wahllokalsystem.ergebnismeldungservice.eai.infomanagement.client.KonfigurierterWahltagControllerApi;
+import de.muenchen.oss.wahllokalsystem.ergebnismeldungservice.eai.infomanagement.model.KonfigurierterWahltagDTO;
+import de.muenchen.oss.wahllokalsystem.ergebnismeldungservice.exception.ExceptionConstants;
+import de.muenchen.oss.wahllokalsystem.ergebnismeldungservice.service.ergebnismeldung.WahlartModel;
+import de.muenchen.oss.wahllokalsystem.ergebnismeldungservice.service.ergebnismeldung.WahlenClient;
+import de.muenchen.oss.wahllokalsystem.wls.common.exception.WlsException;
+import de.muenchen.oss.wahllokalsystem.wls.common.exception.util.ExceptionFactory;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.apache.commons.collections.CollectionUtils;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+@Component
+@Profile(Profiles.NOT + Profiles.DUMMY_CLIENTS)
+@RequiredArgsConstructor
+@Slf4j
+public class BasisdatenClient implements WahlenClient {
+
+    private final KonfigurierterWahltagControllerApi konfigurierterWahltagControllerApi;
+    private final WahlenControllerApi wahlenControllerApi;
+
+    private final ExceptionFactory exceptionFactory;
+
+    private final BasisdatenClientMapper basisdatenClientMapper;
+
+    @Override
+    public WahlartModel getWahlartOfCurrentWahltag(final String wahlID) {
+        val currentWahltagID = getConfiguredWahltagID();
+        val wahlartOfWahl = getWahlen(currentWahltagID, wahlID);
+
+        return basisdatenClientMapper.toModel(wahlartOfWahl);
+    }
+
+    private String getConfiguredWahltagID() {
+        final KonfigurierterWahltagDTO konfigurierterWahltag;
+        try {
+            konfigurierterWahltag = konfigurierterWahltagControllerApi.getKonfigurierterWahltag();
+        } catch (final WlsException wlsException) {
+            log.debug("found WlsException", wlsException);
+            throw wlsException;
+        } catch (final Exception e) {
+            throw exceptionFactory.createTechnischeWlsException(ExceptionConstants.KOMMUNIKATIONSFEHLER_MIT_INFOMANAGEMENT);
+        }
+
+        if (konfigurierterWahltag == null || konfigurierterWahltag.getWahltagID() == null) {
```

Comment:
```suggestion
        if (konfigurierterWahltag == null && konfigurierterWahltag.getWahltagID() == null) {
```

soltle hier kein und stehen? der hintere teil spielt bei nem oder nie ne rolle, da es automatisch null ist, wenn der erste teila auch null ist

Reply:
ein `&&` geht nicht. Wenn `konfigurierterWahltag` null ist, dann würde `konfigurierterWahltag .getWahltagID()` eine NPE werfen.

IntelliJ sagt `Condition 'konfigurierterWahltag. getWahltagID() == null' is always 'false'` was aber nur zutrifft wenn sich alles an die API hält. Ich finde den Check so wie er ist ok um auf der sicheren Seite zu sein.